### PR TITLE
refactor #1531: migrated payments screen to compose

### LINKED ISSUE #1531: Migrate xml to Jetpack compose - home/ui/PaymentsFragment
**Summary:**

Summarize your issue here.

**Steps to reproduce:**

How can we reproduce again the issue?

**Expected behavior:**

What did you expect the app to do?

**Observed behavior:**

What did you see instead?  Describe your issue in detail here.

**Device and Android version:**

What make and model device (e.g., Samsung Galaxy S3) did you encounter this on?  What Android
version (e.g., Android 4.0 Ice Cream Sandwich or Android 6.0 Marshmallow) are you running?  Is it
 the stock
version from the manufacturer or a custom ROM?

**Screenshots:**

Can be created by pressing the Volume Down and Power Button at the same time on Android 4.0 and higher.

### PR DESCRIPTION
**Pull request by PratyushSingh07**
## Issue Fix
Fixes #1531 

https://github.com/openMF/mobile-wallet/assets/90026952/b4fdfe38-dbfc-48cd-b62c-0d59c39a6869

- [x] Apply the `AndroidStyle.xml` style template to your code in Android Studio.

- [x] Run the unit tests with `./gradlew check` to make sure you didn't break anything

- [x] If you have multiple commits please combine them into one commit by squashing them.

### CODE REVIEW

**Comment by PratyushSingh07** [comment]
@therajanmaurya in this PR, I have migrated the SendScreen and RequestScreen to compose in this PR.
However HistoryFragment, SIFragment are yet to be migrated. Can you please reivew this and merge it for the time being. Once merged I will fix the rest of the screens as well.

I am using HorizontalPager here so the rest of HistoryFragment and SIFragment will be blank for the time being.

**Comment by PratyushSingh07** [comment]
Please review this @therajanmaurya . I have migrated the payments screen to compose. Please review and merge this . Once merged I will look for any other bugs and edge cases that I might have missed

File: mifospay/src/main/java/org/mifos/mobilewallet/mifospay/history/presenter/HistoryViewModel.kt [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+
+    init {
+        fetchTransactions()
+    }
+}
+
+sealed class HistoryUiState {
+    data object Initial : HistoryUiState()
+    data object Loading : HistoryUiState()
+    data object EmptyList : HistoryUiState()
```
Comment (therajanmaurya) [inline mifospay/src/main/java/org/mifos/mobilewallet/mifospay/history/presenter/HistoryViewModel.kt]: Empty

File: mifospay/src/main/java/org/mifos/mobilewallet/mifospay/history/presenter/HistoryViewModel.kt [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+    data class Error(val message: String) : HistoryUiState()
+    data class HistoryList(val list: List<Transaction>?) : HistoryUiState()
```
Comment (therajanmaurya) [inline mifospay/src/main/java/org/mifos/mobilewallet/mifospay/history/presenter/HistoryViewModel.kt]: make it default empty and remove null

File: mifospay/src/main/java/org/mifos/mobilewallet/mifospay/history/ui/HistoryItem.kt [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+            Row(
+                modifier = Modifier
+                    .fillMaxWidth()
+                    .padding(end = 10.dp),
+                horizontalArrangement = Arrangement.End
+            ) {
+                Text(
+                    text = transactionType.name,
+                    color = if (transactionType == TransactionType.CREDIT) Color(0xFF009688) else Color(0xFFF10606),
+                    fontSize = 16.sp,
```
Comment (therajanmaurya) [inline mifospay/src/main/java/org/mifos/mobilewallet/mifospay/history/ui/HistoryItem.kt:52]: use typography

File: mifospay/src/main/java/org/mifos/mobilewallet/mifospay/history/ui/HistoryItem.kt [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+            )
+            Row(
+                modifier = Modifier
+                    .fillMaxWidth()
+                    .padding(end = 10.dp),
+                horizontalArrangement = Arrangement.End
+            ) {
+                Text(
+                    text = transactionType.name,
+                    color = if (transactionType == TransactionType.CREDIT) Color(0xFF009688) else Color(0xFFF10606),
```
Comment (therajanmaurya) [inline mifospay/src/main/java/org/mifos/mobilewallet/mifospay/history/ui/HistoryItem.kt]: move to color

File: mifospay/src/main/java/org/mifos/mobilewallet/mifospay/history/ui/HistoryScreen.kt [hunk 5]
```diff
@@ -0,0 +1,136 @@
...
+import com.mifos.mobilewallet.model.domain.TransactionType
+import org.mifos.mobilewallet.mifospay.R
+import org.mifos.mobilewallet.mifospay.designsystem.component.MifosErrorLayout
+import org.mifos.mobilewallet.mifospay.designsystem.component.MifosLoadingWheel
+import org.mifos.mobilewallet.mifospay.designsystem.theme.chipSelectedColor
+import org.mifos.mobilewallet.mifospay.designsystem.theme.lightGrey
+import org.mifos.mobilewallet.mifospay.history.presenter.HistoryUiState
+
+@Composable
+fun HistoryScreen(
```
Comment (therajanmaurya) [inline mifospay/src/main/java/org/mifos/mobilewallet/mifospay/history/ui/HistoryScreen.kt:54]: HistoryScreenContent

File: mifospay/src/main/java/org/mifos/mobilewallet/mifospay/home/ui/PaymentsFragment.kt [hunk 6]
```diff
@@ -1,79 +1,165 @@
...
-        tabLayoutAdapter.addFragment(SIFragment(), getString(R.string.standing_instruction))
-        tabLayoutAdapter.addFragment(InvoicesFragment(), getString(R.string.invoices))
-        vpTabLayout!!.adapter = tabLayoutAdapter
+    private fun showQrClicked() {
+        val intent = Intent(activity, ShowQrActivity::class.java)
+        intent.putExtra(Constants.QR_DATA, transferPresenter.vpa.value)
+        startActivity(intent)
+    }
+
+    private fun searchContactClicked() {
```
Comment (therajanmaurya) [inline mifospay/src/main/java/org/mifos/mobilewallet/mifospay/home/ui/PaymentsFragment.kt]: Move this logic to compose

File: mifospay/src/main/java/org/mifos/mobilewallet/mifospay/home/ui/PaymentsFragment.kt [hunk 7]
```diff
@@ -1,79 +1,165 @@
...
+            val intent = Intent(
+                Intent.ACTION_PICK,
+                ContactsContract.CommonDataKinds.Phone.CONTENT_URI
+            )
+            startActivityForResult(intent, PICK_CONTACT)
+        }
+    }
+
+    private fun scanQrClicked() {
+        if (ContextCompat.checkSelfPermission(
```
Comment (therajanmaurya) [inline mifospay/src/main/java/org/mifos/mobilewallet/mifospay/home/ui/PaymentsFragment.kt]: move this permission to compose

File: mifospay/src/main/java/org/mifos/mobilewallet/mifospay/home/ui/PaymentsScreen.kt [hunk 8]
```diff
@@ -0,0 +1,103 @@
...
+        }
+
+        HorizontalPager(
+            state = pagerState,
+            modifier = Modifier.weight(1f)
+        ) { page ->
+            when (page) {
+                0 -> SendScreen({ scanQr.invoke() }, { searchContact.invoke() }, {})
+                1 -> RequestScreen(showQr = { showQr.invoke() })
+                2 -> HistoryScreen(historyViewModel.historyUiState.collectAsStateWithLifecycle().value)
```
Comment (therajanmaurya) [inline mifospay/src/main/java/org/mifos/mobilewallet/mifospay/home/ui/PaymentsScreen.kt]: name field and use

File: mifospay/src/main/java/org/mifos/mobilewallet/mifospay/home/ui/PaymentsScreen.kt [hunk 9]
```diff
@@ -0,0 +1,103 @@
...
+                3 -> SIScreen(
+                    standingInstructionViewModel.standingInstructionsUiState.collectAsStateWithLifecycle().value,
```
Comment (therajanmaurya) [inline mifospay/src/main/java/org/mifos/mobilewallet/mifospay/home/ui/PaymentsScreen.kt]: make field and use

**Review by therajanmaurya — CHANGES_REQUESTED** [review]

**Pull request merged**